lib/data/repository/movies_from_api.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+
+import 'package:dartz/dartz.dart';
+import 'package:http/http.dart' as http;
+
+import '../../domain/entity/movie.dart';
+import '../../domain/repository/i_movie_repository.dart';
+import '../../domain/repository/service/i_movie_service.dart';
+import '../../domain/use_cases/use_case_interface.dart';
+import '../data_sources/mappers/api_dto.dart';
+import '../data_sources/remote/api_const.dart';
+import '../data_sources/remote/api_service.dart';
+import '../model/model.dart';
+
+class MoviesFromAPI extends MovieRepository {
+  final MovieService movieService;
+  const MoviesFromAPI({this.movieService = const APIMovieServiceImpl()});
+
+  @override
+  Future<EitherMovieAPI<List<MovieEntity>>> getMovies(EndPoint endPoint) async {
+    http.Response res;
```

Comment:
remember to use full descriptive names, you can use response in here, the extra characters are worth the code been more understandable